src/silo/query_engine/filter_expression/negation.cpp

```diff
@@ -31,7 +31,7 @@ std::unique_ptr<operators::Operator> Negation::compile(
       return child_operator;
    }
    if (child_operator->type() == operators::SELECTION) {
-      dynamic_cast<operators::Selection*>(child_operator.get())->negate();
+      dynamic_cast<operators::Selection<uint64_t>*>(child_operator.get())->negate();
```

Comment:
This is a little hacky, since we need to cast the operator to some selection where the generic type does not matter...

Reply:
I agree and in the long run we probably need something better but did you test this? I would expect the `dynamic_cast` to return a nullptr if the selection is not of type `uint64_t`

Reply:
It does work.

```
{
  "action": {
    "type": "Aggregated"
  },
  "filterExpression": {
    "type": "Not",
    "child": {
      "type": "StringEquals",
      "column": "division",
      "value": "Solothurn"
    }
  }
}
```
returns the expected result.